### PR TITLE
[WFCORE-6423] Allow reload operation to send an early response when u…

### PR DESCRIPTION
**Pull request by yersan**
…sing a remote proxy controller

Jira issue: https://issues.redhat.com/browse/WFCORE-6423

### CODE REVIEW

**Comment by yersan** [comment]
@bstansberry I've added some documentation about how is the TransactionalProtocolClient implementation as suggested in the Jira. 

From my point of view, `TransactionalProtocolHandlers#executeBlocking` should be more of a TransactionalProtocolClient responsibility instead of `TransactionalProtocolHandlers`

File: controller/src/main/java/org/jboss/as/controller/remote/TransactionalProtocolOperationHandler.java [hunk 1]
```diff
@@ -75,6 +88,7 @@
  * @author <a href="mailto:darran.lofthouse@jboss.com">Darran Lofthouse</a>
  */
 public class TransactionalProtocolOperationHandler implements ManagementRequestHandlerFactory {
+    private static final Set<String> PREPARED_RESPONSE_OPERATIONS = new HashSet<>(List.of(RELOAD));
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/remote/TransactionalProtocolOperationHandler.java:91]: This should be an immutable set.

**Review by bstansberry — CHANGES_REQUESTED** [review]
I have one minor change suggestion.

This seems like something that should wait for WF 30, i.e. merge it next week after the core 21.1 release. I think it's fine but it doesn't seem like something so urgent that it should be added to 29 at the last moment.

File: controller/src/main/java/org/jboss/as/controller/remote/TransactionalProtocolOperationHandler.java (hunk 1 above)
Comment (yersan) [reply controller/src/main/java/org/jboss/as/controller/remote/TransactionalProtocolOperationHandler.java]: Done

Comment (yersan) [reply controller/src/main/java/org/jboss/as/controller/remote/TransactionalProtocolOperationHandler.java]: @bstansberry Could you take another look?

**Review by bstansberry — APPROVED** [review]

**Comment by yersan** [comment]
Tnanks @bstansberry

**Pull request merged**